tests/TESTINS5.MLC

```diff
@@ -341,6 +374,221 @@ BIC_OK2  EQU    *
+         LM     R5,R6,=F'-2,-3'
+         MSRKC  R2,R5,R6
+         CL     R2,=F'6'
+         RT1    CC0 
+         LM   R5,R6,=XL4'7FFFFFFF,2'
+         MSRKC  R2,R5,R6
+         RT1    CC3  
+* MSGRKC
+         LMG   R5,R6,=FL8'2,-3'
+         L	R2,=F'2'
+         SR       R0,R0
+         MSGRKC  R2,R5,R6
+         RT1 CC2
+         CLG    R2,=FL8'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'2,-3'
+         SR       R0,R0
+         MSGRKC  R2,R5,R6
+         RT1    CC1
+         CLG    R2,=FL8'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,3'
+         SR       R0,R0
+         MSGRKC  R2,R5,R6
+         RT1     CC1
+         CLG    R2,=FL8'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,-3'
+         SR       R0,R0
+         MSGRKC  R2,R5,R6
+         RT1   CC2
+         CLG   R2,=FL8'6'
+         RT1    CC0 
+         LMG   R5,R6,=XL8'7FFFFFFFFFFFFFFF,2'
+         SR       R0,R0
+         MSGRKC  R2,R5,R6
+          RT1    CC3  
+* MSRKC #300
+         LM    R5,R6,=F'2,3'
+         SR     R0,R0
+         MSRKC  R2,R5,R6
+         RT1    CC2
+         CL      R2,=F'6'
+         RT1    CC0 
+         LM     R5,R6,=F'2,-3'
+         SR      R0,R0
+         MSRKC  R2,R5,R6
+         RT1   CC1
+         CL      R2,=F'-6'
+         RT1    CC0 
+         LM     R5,R6,=F'-2,3'
+         SR      R0,R0
+         MSRKC  R2,R5,R6
+         RT1    CC1
+         CL      R2,=F'-6'
+         RT1    CC0 
+         LM     R5,R6,=F'-2,-3'
+         SR      R0,R0
+         MSRKC  R2,R5,R6
+         RT1   CC2
+         CL     R2,=F'6'
+         RT1    CC0 
+         LM   R5,R6,=XL4'7FFFFFFF,2'
+         SR       R0,R0
+         MSRKC  R2,R5,R6
+         RT1    CC3  
+* MSC #300 
+         L	R2,=F'2'
+         SR       R0,R0
+	MSC R2,=F'3'
+          RT1  CC2
+          CL    R2,=F'6'
+	RT1	CC0
+         L	R2,=F'2'
+         SR       R0,R0
+	MSC R2,=F'-3'
+          RT1  CC1
+          CL    R2,=F'-6'
+         RT1  CC0
+         L	R2,=F'-2'
+         SR       R0,R0
+	MSC R2,=F'3'
+          RT1  CC1
+          CL    R2,=F'-6'
+	RT1	CC0
+         L	R2,=F'-2'
+         SR       R0,R0
+	MSC R2,=F'-3'
+          RT1  CC2
+          CL    R2,=F'6'
+	RT1	CC0
+         L	R2,=X'7FFFFFFF'
+	MSC R2,=F'2'
+          RT1  CC3
+* MSGC
+         LG	R2,=FL8'2'
+         SR       R0,R0
+	MSC R2,=F'2'
+          RT1  CC3
+* MSGC #300
+         LG	R2,=FL8'2'
+         SR       R0,R0
+	MSGC R2,=FL8'3'
+          RT1  CC2
+          CLG    R2,=FL8'6'
+	RT1	CC0
+         LG	R2,=FL8'2'
+         SR       R0,R0
+	MSGC R2,=FL8'-3'
+          RT1  CC1
+          CLG    R2,=FL8'-6'
+         RT1  CC0
+         LG	R2,=FL8'-2'
+         SR       R0,R0
+	MSGC R2,=FL8'3'
+          RT1  CC1
+          CLG    R2,=FL8'-6'
+	RT1	CC0
+         LG	R2,=FL8'-2'
+         SR       R0,R0
+	MSGC R2,=FL8'-3'
+          RT1  CC2
+          CLG    R2,=FL8'6'
+	RT1	CC0
+         LG	R2,=X'7FFFFFFFFFFFFFFF'
+         SR       R0,R0
+	MSGC R2,=FL8'2'
+          RT1  CC3
```

Comment:
Same comments about overflow. Should still check that R2 contains the expected value.

Also add tests for product equal to zero.